.circleci/config.yml

```diff
@@ -543,6 +543,21 @@ jobs:
       - checkout
       - setup_environment:
           platform: << parameters.platform >>
+      - when:
+          condition:
+            not: << pipeline.parameters.nightly >>
+          steps:
+            - run:
+                command: >
+                  # Validate that version in the Cargo.toml matches the tag version
+                  # Fail the release if the versions don't match
+                  # Note that the tag has the prefix `v`, so this needs to be used to prefix the VERSION variable.
+                  VERSION=v$(cargo metadata --format-version=1 --no-deps | jq --raw-output '.packages[0].version')
+                  if [[ "${VERSION}" != "${CIRCLE_TAG}" ]]; then
+                    echo "Version in Cargo.toml (${VERSION}) does not match tag (${CIRCLE_TAG})"
+                    exit 1
+                  fi
+
```

Comment:
This will run X separate runs for each build of the same release.  Is there any easy way we could make this run in a pre-flight job in front of all the release jobs?  We have several things we need to check, and it would make sense to club them together.

Reply:
If the answer is "yes, but let's move it out later", then that's fine.  I just started introducing the "Existing image" check last week, and it would be best positioned _before_ starting 4 (somewhat expensive) jobs.